src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -260,66 +875,87 @@ static std::tuple<bool,        // is suitable kernel found
         }
         if(!gemm_k_valid)
             continue;
+
+        // the following check is only used for bwd fp16
+        if(ctx.IsFp16())
+        {
+            // output vector load limitation, n1b
+            if(cfg.nxe == 0 && cfg.tensor_b_thread_lengths[3] > 1 &&
+               (ho * wo) % cfg.tensor_b_thread_lengths[3] != 0)
+                continue;
+
+            bool is_unit_yx = (y == 1 && x == 1);
+
+            // configs with nxe == 1 and tensor_a_thread_lengths[3] > 1 can only be used with x=y=1
+            if(cfg.nxe == 1 && cfg.tensor_a_thread_lengths[3] > 1 && !is_unit_yx)
+                continue;
+
+            // let's check the next configuration even though this configuration is applicable
+            if(mayHaveBiggerGemmNPerBlock(gemm_m, gemm_n, is_unit_yx, cfg))
+                continue;
+        };
+
         return std::make_tuple(true,
+                               cfg,
                                cfg.GetKernelName(),
                                cfg.GetBlockSize(),
-                               integer_divide_ceil(gemm_m, cfg.gemm_m_per_block) *
-                                   integer_divide_ceil(gemm_n, cfg.gemm_n_per_block));
+                               group * integer_divide_ceil(gemm_m, cfg.gemm_m_per_block) *
+                                   integer_divide_ceil(gemm_n_packed, cfg.gemm_n_per_block));
     }
 
-    // second try, try find if packed image size match
-    for(const auto& cfg : tunables)
+    if(ctx.IsFp32())
     {
-        const auto b = cfg.nxe == 0
-                           ? h_tilda_slice * w_tilda_slice
-                           : ((h_tilda_slice * w_tilda_slice + cfg.nxb - 1) / cfg.nxb) * cfg.nxb;
-        const auto gemm_n_packed = n * b;
-        if(cfg.nxe == 0)
+        // second try, try find if packed image size match
```

Comment:
`gemm_n_packed` has been check before this piece of code, why it need a second try?

Reply:
My commit does not  touch the stuff used by fp32,  so in the first try, gemm_n_packed is the same as gemm_n